### PR TITLE
Stops gibber from shooting gibs through walls

### LINKED ISSUE #20728: Gibber leaves a mess through walls
#### Description of issue

When using the gibber, walls blocking its output direction do not block bloody messes.
image
In this image, a human was gibbed. Blood appeared in the maintenance tunnel afterwards (see highlighted tile).

I only saw this happen just now on Roidstation - it may not happen on other maps where the gibber is placed further from a wall, and I'm not sure if the distance bloody messes are produced is chance-based.
May permit metagay in knowing if the gibber has been used or not in disposal of a corpse, although this probably happens for legitimate uses of the gibber too

#### Difference between expected and actual behavior

Blood should not appear behind walls that the gibber is pointing towards.

#### Steps to reproduce

Have a gibber one tile away from a wall (this may also happen if the gibber is right against the wall, untested)
Activate gibber
Check behind the wall for blood

#### Specific information for locating

Gibber on Roidstation.
Likely occurs with any gibber on any map placed appropriately in relation to a wall.

#### Length of time in which bug has been known to occur

Just noticed today.

#### Server revision

Game: fad54ddedf4b7c7d79babc910c4caf8f45e24f73 
Patches: 6a64204fc9b9cce851f966cf3cbd302ab482f846 

<general> <roid> <exploitable>

### PR DESCRIPTION
**Pull request by veles2001**
Closes #20728

### CODE REVIEW

File: code/game/machinery/kitchen/gibber.dm [hunk 1]
```diff
@@ -235,15 +235,22 @@ obj/machinery/gibber/New()
...
 			if (!Tx.density)
-				var/obj/effect/decal/cleanable/blood/gibs/O = getFromPool(/obj/effect/decal/cleanable/blood/gibs, Tx)
-				O.New(Tx,i)
+				if(!no_more_gibs)
+					var/obj/effect/decal/cleanable/blood/gibs/O = getFromPool(/obj/effect/decal/cleanable/blood/gibs, Tx)
+					O.New(Tx,i)
+			else
+				no_more_gibs = TRUE
+				if(i == 1)
+					var/obj/effect/decal/cleanable/blood/gibs/O = getFromPool(/obj/effect/decal/cleanable/blood/gibs, locate(src.x, src.y, src.z))
```
Comment (Exxion) [inline code/game/machinery/kitchen/gibber.dm]: You can replace the whole `locate()` call with `get_turf(src)`

File: code/game/machinery/kitchen/gibber.dm [hunk 2]
```diff
@@ -235,15 +235,22 @@ obj/machinery/gibber/New()
...
+					O.New(locate(src),i)
```
Comment (Exxion) [inline code/game/machinery/kitchen/gibber.dm]: Same here

File: code/game/machinery/kitchen/gibber.dm [hunk 3]
```diff
@@ -235,15 +235,22 @@ obj/machinery/gibber/New()
...
 		for (var/i=1 to totalslabs)
 			var/obj/item/meatslab = allmeat[i]
 			var/turf/Tx = locate(src.x - i, src.y, src.z)
 			meatslab.forceMove(src.loc)
 			meatslab.throw_at(Tx,i,3)
 			if (!Tx.density)
-				var/obj/effect/decal/cleanable/blood/gibs/O = getFromPool(/obj/effect/decal/cleanable/blood/gibs, Tx)
-				O.New(Tx,i)
+				if(!no_more_gibs)
+					var/obj/effect/decal/cleanable/blood/gibs/O = getFromPool(/obj/effect/decal/cleanable/blood/gibs, Tx)
```
Comment (MadmanMartian) [inline code/game/machinery/kitchen/gibber.dm]: After having looked at the getFromPool proc, everything after the first argument is passed to the first arguments New, so calling O.New is pointless.

Change it to getFromPool(/obj/effect/decal/cleanable/blood/gibs, Tx, i) and remove the O.new(...), same with the one below (Except have it be getFromPool(/obj/effect/decal/cleanable/blood/gibs, locate(src), i) instead)

**Comment by veles2001** [comment]
Neato, I'll make those changes real quick

**Review by MadmanMartian — CHANGES_REQUESTED** [review]

File: code/game/machinery/kitchen/gibber.dm (hunk 3 above)
Comment (Exxion) [reply code/game/machinery/kitchen/gibber.dm]: Not `locate(src)` since that doesn't make sense

**Comment by veles2001** [comment]
Does everything look good now?
And sorry if my code is a bit weird, I'm not used to BYOND so I've been trying get some practice by fixing small issues.

**Review by MadmanMartian — APPROVED** [review]

**Review by MadmanMartian — COMMENTED** [review]
Why not have it start at i = 0, and if TX.density, break?

This would spawn a gib on its turf, and still stop at a dense turf

**Comment by veles2001** [comment]
But wouldn't breaking spawn an inadequate amount of meat?

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
Inadequate Amount of Meat is my new band name

**Comment by MadmanMartian** [comment]
You're right

**Comment by MadmanMartian** [comment]
Been deliberating on this, and had a thought.

Why does it only go right? Why not in all directions? Same with the meat

**Comment by DamianX** [comment]
It goes left, not right. And it's probably because the gibber was only sprited facing left.

**Pull request merged**